rsciio/digital_micrograph/_api.py

```diff
@@ -1237,27 +1238,31 @@ def get_mapping(self):
         return mapping
 
 
-def file_reader(filename, record_by=None, order=None, lazy=False, optimize=True):
+def file_reader(
+    filename, lazy=False, record_by=None, order=None, optimize=True, **kwargs
+):
     """Reads a DM3/4 file and loads the data into the appropriate class.
-    data_id can be specified to load a given image within a DM3/4 file that
-    contains more than one dataset.
+    If more than one dataset is contained in the ``.dm3/4`` file, a list of
+    signals is returned.
 
     Parameters
     ----------
-    record_by: Str
+    %s
+    %s
+    record_by: str
```

Comment:
This may be something to follow up in a separate PR.

Reply:
Giving more details?

Reply:
Yes, maybe or whether, this is still used or not.

Reply:
It is saved into the metadata, but DM is the only reader that allows setting it as parameter. Normally, it should be determined from the metadata. So I will also remove it from here.

Values can be "spectrum" or "image".

Reply:
[`record_by` is deprecated since HyperSpy 1.2](https://hyperspy.readthedocs.io/en/latest/user_guide/metadata_structure.html#signal), maybe it should also be removed altogether from the metadata dictionnary returned by the reader?

Reply:
OK, then we should open a separate issue and PR, because there are still several readers setting this parameter!

Reply:
see  #79